src/security/KeyData.cc

```diff
@@ -96,14 +96,34 @@ Security::KeyData::loadX509ChainFromFile()
         return;
     }
 
-    if (X509_check_issued(cert.get(), cert.get()) == X509_V_OK)
-        debugs(83, 5, "Certificate is self-signed, will not be chained");
-    else {
+    if (X509_check_issued(cert.get(), cert.get()) == X509_V_OK) {
+        char *nameStr = X509_NAME_oneline(X509_get_subject_name(cert.get()), nullptr, 0);
+        debugs(83, DBG_IMPORTANT, "Certificate is self-signed, will not be chained: " << nameStr);
```

Comment:
We have had quite a few user questions and some confusion about why PEM files are not working the way they expected where it turned out to be misconfiguration with self-signed certs or incorrect chain ordering. These debugs are added (and at that level specifically) to make it clear when the problem is a misconfiguration and what certs Squid is actually loading.

Reply:
We cannot successfully address admin confusion by polluting logs in benign cases. If Squid can detect something unusual and/or potentially dangerous, sure, let's complain at level 1! Otherwise, excessive logging will only _hide_ important information. In this particular case, if, for example, 95+% of chains are going to contain a self-signed certificate (without any harm!), then we should not log about that boring fact at level 1.

Reply:
very well. Dropped these into parse notes.